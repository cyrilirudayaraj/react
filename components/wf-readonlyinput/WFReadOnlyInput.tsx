import React from 'react';
import { ReadOnlyInput, FormField } from '@athena/forge';
import StringUtil from '../../utils/StringUtil';

const WFReadOnlyInput = (props: PropTypes): JSX.Element => {
  return (
    <div className="readonlywrapper">
      <FormField
        id={props.id || props.labelText}
        inputAs={ReadOnlyInput}
        labelText={props.labelText}
        text={
          props.href || props.children
            ? ''
            : StringUtil.returnHyphenIfEmpty(props?.text)
        }
      >
        {props.children}
        {!props.children && props.href && props.text && (
          <p>
            <a href={props.href} target={props.target || ''}>
              {props.text}
            </a>
          </p>
        )}
      </FormField>
    </div>
  );
};

interface PropTypes {
  labelText: string;
  text?: string | null | undefined;
  href?: string;
  target?: string;
  children?: React.ReactNode;
  id?: string;
}
export default WFReadOnlyInput;
