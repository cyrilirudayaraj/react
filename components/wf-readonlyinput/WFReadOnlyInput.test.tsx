import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WFReadOnlyInput from './WFReadOnlyInput';

configure({ adapter: new Adapter() });
let wrapper: any;

describe('snapshot test <WFReadOnlyInput>', () => {
  it('should render as label', () => {
    wrapper = shallow(<WFReadOnlyInput labelText="Patient Name" text="Roby" />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render as anchor', () => {
    wrapper = shallow(
      <WFReadOnlyInput labelText="Rule Id" text="1.1233" showAsLink={true} />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
