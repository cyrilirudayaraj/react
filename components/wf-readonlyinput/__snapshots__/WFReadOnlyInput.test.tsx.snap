// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <WFReadOnlyInput> should render as anchor 1`] = `
"<div className=\\"readonlywrapper\\">
  <FormField id=\\"Rule Id\\" inputAs={[Function: ReadOnlyInput]} labelText=\\"Rule Id\\" text=\\"1.1233\\" required={false} />
</div>"
`;

exports[`snapshot test <WFReadOnlyInput> should render as label 1`] = `
"<div className=\\"readonlywrapper\\">
  <FormField id=\\"Patient Name\\" inputAs={[Function: ReadOnlyInput]} labelText=\\"Patient Name\\" text=\\"Roby\\" required={false} />
</div>"
`;
